### PR TITLE
Add proper remote_name to gh-deploy config

### PR DESCRIPTION
**Pull request by itsayellow**
Simple addition of `remote_name` in mkdocs.yml config, so gh-deploy will always push to the pipxproject/pipx remote.

### CODE REVIEW

**Comment by itsayellow** [comment]
I tried to make a full url as the `remote_name` but got a strange error, so I changed it to `upstream`.

Here's the error that I got when I tried to make `remote_name` the literal url `git@github.com:pipxproject/pipx.git`:
```
nox > mkdocs gh-deploy
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /Users/mclapp/git/pipx/site 
WARNING -  Version check skipped: No version specificed in previous deployment. 
INFO    -  Copying '/Users/mclapp/git/pipx/site' to 'gh-pages' branch and pushing to GitHub. 
ERROR   -  Failed to deploy to GitHub with error: 
To github.com:pipxproject/pipx.git
 ! [rejected]        gh-pages -> gh-pages (non-fast-forward)
error: failed to push some refs to 'git@github.com:pipxproject/pipx.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

I verified it works fine if I use `upstream` as the remote_name instead.  This just means that `upstream` remote will need to be configured properly for `mkdocs gh-deploy` to work.

**Comment by itsayellow** [comment]
It would be nice to use the literal url to push docs, maybe somebody with more git experience knows why it didn't work?

**Comment by cs01** [comment]
Can you share the output of 
```
>> git remote -v
```
on your machine?

For me it's
```
origin	git@github.com:pipxproject/pipx.git (fetch)
origin	git@github.com:pipxproject/pipx.git (push)
```

I think if you have this set, you should be fine. I'm guessing yours is going to shown a name of `upstream` rather than `origin`. If that is the case, you can run
```
git remote set-url origin git@github.com:pipxproject/pipx.git
```
or just wipe the directory and re-clone, which should set the origin appropriately.

**Comment by cs01** [comment]
mkdocs just ends up calling `git push <remote> <branch>`, so I don't think we have a way around this. The name `origin` is the most canonical and should be the default name when you clone a repo, so I would like to stick with that.

https://github.com/mkdocs/mkdocs/blob/273e3baa8e47fa5d121402d98757d3f5a5510383/mkdocs/utils/ghp_import.py#L169

**Comment by cs01** [comment]
Creating a GitHub action rather than running this command locally would probably be a better long-term solution. Same with publishing to PyPI.

**Comment by itsayellow** [comment]
> Can you share the output of
> 
> ```
> >> git remote -v
> ```
> 
> on your machine?
> 
> For me it's
> 
> ```
> origin	git@github.com:pipxproject/pipx.git (fetch)
> origin	git@github.com:pipxproject/pipx.git (push)
> ```
> 
> I think if you have this set, you should be fine. I'm guessing yours is going to shown a name of `upstream` rather than `origin`. If that is the case, you can run
> 
> ```
> git remote set-url origin git@github.com:pipxproject/pipx.git
> ```
> 
> or just wipe the directory and re-clone, which should set the origin appropriately.

My origin is my fork.  pipxproject is where I submit PR's to (upstream)
```
carvel:pipx git remote -v
origin	git@github.com:itsayellow/pipx.git (fetch)
origin	git@github.com:itsayellow/pipx.git (push)
upstream	git@github.com:pipxproject/pipx.git (fetch)
upstream	git@github.com:pipxproject/pipx.git (push)
```

I thought that you (@cs01) are the only person who can use pipxproject/pipx.git as origin, since pipxproject is your account?  Or at the very least I thought the desired flow was, develop on your own fork and then PR to the main one from your fork.

**Comment by itsayellow** [comment]
OK then, I guess I can use pipxproject/pipx.git as origin and successfully push to it (just tried it.)

Is this the desired flow for maintainers?  I was sticking with my fork as "good practice", but I can change if it's better we all use the same project repo.

**Comment by cs01** [comment]
Changing the origin of the git repo doesn't really do much. You can still push to your fork, you'll just have to give it a different name in git, and then push to it `git push myfork mybranch`.

I don't really think it's better or worse all sharing the same repo. It's just a different convention. I don't mind if maintainers push branches to this repo, but I don't want to end up in a situation where the number of branches is overwhelming and they aren't cleaned up after they're merged. 

Bottom line: feel free to push where ever you want, but if you push to pipx directly just be sure to clean up branches after they're no longer needed (which I am pretty sure you do anyway).

**Comment by itsayellow** [comment]
In that case, probably we should just close this pull request without merging.

**Comment by cs01** [comment]
Agreed
